exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java

```diff
@@ -165,32 +169,59 @@ public DrillbitContext getContext() {
    *
    * <p>This is intended to be used by {@link org.apache.drill.exec.server.Drillbit#close()}.</p>
    */
-  public void waitToExit() {
+  public void waitToExit(Drillbit bit, boolean forcefulShutdown) {
     synchronized(this) {
-      if (queries.isEmpty() && runningFragments.isEmpty()) {
+      numOfRunningQueries = queries.size();
+      numOfRunningFragments = runningFragments.size();
+      if ( queries.isEmpty() && runningFragments.isEmpty()) {
```

Comment:
We just updated two variables with counts, then we ignore them. Do we even need them?

Reply:
We compare if the remaining queries are going down or if new queries are added during shutdown.